engine/api/application_variable.go

```diff
@@ -29,45 +30,55 @@ func (api *API) getVariablesAuditInApplicationHandler() service.Handler {
 
 func (api *API) getVariableAuditInApplicationHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
-		// Get project name in URL
 		vars := mux.Vars(r)
-		key := vars[permProjectKey]
+		projectKey := vars[permProjectKey]
 		appName := vars["applicationName"]
 		varName := vars["name"]
 
-		app, errA := application.LoadByName(api.mustDB(), key, appName)
-		if errA != nil {
-			return sdk.WrapError(errA, "getVariableAuditInApplicationHandler> Cannot load application %s on project %s", appName, key)
+		proj, err := project.Load(api.mustDB(), projectKey)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load project %s", projectKey)
+		}
+
+		app, err := application.LoadByProjectIDAndName(ctx, api.mustDB(), proj.ID, appName)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load application %s on project %s", appName, proj.Key)
 		}
 
-		variable, errV := application.LoadVariable(api.mustDB(), app.ID, varName)
-		if errV != nil {
-			return sdk.WrapError(errV, "getVariableAuditInApplicationHandler> Cannot load variable %s", varName)
+		variable, err := application.LoadVariable(ctx, api.mustDB(), app.ID, varName)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load variable %s", varName)
 		}
 
-		audits, errA := application.LoadVariableAudits(api.mustDB(), app.ID, variable.ID)
-		if errA != nil {
-			return sdk.WrapError(errA, "getVariableAuditInApplicationHandler> Cannot load audit for variable %s", varName)
+		audits, err := application.LoadVariableAudits(api.mustDB(), app.ID, variable.ID)
+		if err != nil {
+			return sdk.WrapError(err, "cannot load audit for variable %s", varName)
 		}
+
 		return service.WriteJSON(w, audits, http.StatusOK)
 	}
 }
 
 func (api *API) getVariableInApplicationHandler() service.Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
-		key := vars[permProjectKey]
+		projectKey := vars[permProjectKey]
 		appName := vars["applicationName"]
 		varName := vars["name"]
 
-		app, err := application.LoadByName(api.mustDB(), key, appName)
+		proj, err := project.Load(api.mustDB(), projectKey)
```

Comment:
I think it's a bit sad to load the project to get its ID, whereas the projectKey is an identifier.
we should consider `application.LoadByProjectKeyAndName`